sdcm/sct_runner.py

```diff
@@ -589,9 +601,127 @@ def _get_base_image(self, image: Optional[Any] = None) -> Any:
         return self._get_image_url(image.id)
 
 
+class AzureSctRunner(SctRunner):
+    GALLERY_IMAGE_NAME = "sct-runner"
+    GALLERY_IMAGE_VERSION = f"{SctRunner.VERSION}.0"  # Azure requires to have it in `X.Y.Z' format
+    BASE_IMAGE = {
+        "publisher": "canonical",
+        "offer": "0001-com-ubuntu-server-focal",
+        "sku": "20_04-lts-gen2",
+        "version": "latest",
+    }
+    SOURCE_IMAGE_REGION = AzureRegion.SCT_GALLERY_REGION
+    IMAGE_BUILDER_INSTANCE_TYPE = "Standard_D2s_v3"
+    REGULAR_TEST_INSTANCE_TYPE = "Standard_D2s_v3"  # 2 vcpus, 8G, recommended by Ubuntu 20.04 LTS image publisher
+    LONGTERM_TEST_INSTANCE_TYPE = "Standard_E2s_v3"  # 2 vcpus, 16G, recommended by Ubuntu 20.04 LTS image publisher
+
+    def __init__(self, region_name: str, cloud_provider: str = "azure"):
+        super().__init__(cloud_provider=cloud_provider, region_name=region_name)
+        self.azure_region = AzureRegion(region_name=region_name)
+        self.azure_region_source = AzureRegion(region_name=self.SOURCE_IMAGE_REGION)
+        self.azure_service = self.azure_region.azure_service
+
+    def region_az(self, region_name: str, availability_zone: str) -> str:
+        return region_name
+
+    @cached_property
+    def image_name(self) -> str:
+        return f"sct-runner-{self.VERSION}"
+
+    @cached_property
+    def key_pair(self) -> SSHKey:
+        return KeyStore().get_gce_ssh_key_pair()  # scylla-test
```

Comment:
Is it intentional that we use the same key pair?

Reply:
Yes, I don't see any good reason to add another key pair to the QA keystore.

Reply:
currently, let's use the same key, but in future we will create a new key for each test run (task planned for next sprints)